src/api/Interpret.cc

```diff
@@ -1285,14 +1285,24 @@ void Interpret::getProof()
 
 void Interpret::getUnsatCore() {
     auto const unsatCore = main_solver->getUnsatCore();
-    auto const & unsatCoreTerms = unsatCore->getTerms();
     std::cout << "( ";
     auto & termNames = main_solver->getTermNames();
-    for (PTRef fla : unsatCoreTerms) {
-        if (termNames.contains(fla)) {
+    if (not config.print_cores_full()) {
+        for (PTRef fla : unsatCore->getNamedTerms()) {
+            assert(termNames.contains(fla));
             auto const & name = termNames.nameForTerm(fla);
             std::cout << name << ' ';
         }
+    } else {
+        for (PTRef fla : unsatCore->getTerms()) {
+            if (termNames.contains(fla)) {
+                std::cout << termNames.nameForTerm(fla);
+            }
+            else {
+                std::cout << logic->printTerm(fla);
+            }
+            std::cout << ' ';
+        }
```

Comment:
Why is this needed?

Reply:
Just (slight) performance reasons. In the first branch, we use named terms so all of them must be in `termNames`. In second branch, we iterate over all terms so not all of them are necessarily in `termNames`. If it is not, we just print out string representation of the term, because we do not have any name for it.

Reply:
This commit implements option `:print-cores-full` from https://cvc5.github.io/blog/2024/04/15/interfaces-for-understanding-cvc5.html

Reply:
I see, can you please document the meaning of this option? And maybe also the `minimal-unsat-cores`, to make it clear that these are subset minimal, and not global minimum.

Reply:
Moved to #771